### ISSUE TITLE
ImageServiceProvider.php line 42: Call to undefined method Intervention\Image\ImageManager::pattern()

### ISSUE THREAD

**Issue by MatanYadaev** [Folkloreatelier/laravel-image#47]
```
In ImageServiceProvider.php line 42:
                                                                       
  Call to undefined method Intervention\Image\ImageManager::pattern()  
```
Laravel 5.7, when running `composer update`

**Comment by essteffan** [comment]
I have the same issue, also In laravel 5.7

Any news?

**Comment by aharabara** [comment]
I have a response for you guys. I suppose you use some package like `spatie/browsershot`, `spatie/image` or other stuff that is loading package `intervention/image` which is rewriting `ImageManager::class` singleton in `$app['image']` and when `folklore/image` package is initializing it is calling `$app['image']` which is a totally different object.
Basically `folklore/image` and `intervention/image` packages are conflicting
@essteffan, @MatanYed

**Comment by essteffan** [comment]
You are right @aharabara I am using both of them in a project...but how did it work before?

Best regards!

**Comment by aharabara** [comment]
You did package/project version upgrade. For me it was Laravel update from 5.2 to 5.6. I replaced less problematic package with custom service and it worked 😁 @essteffan

**Comment by Bronskiy** [comment]
I found a slightly weird solution but it works(laravel 5.7) :)
Change "extra' in your composer.json:
 ```
 "extra": {
        "laravel": {
            "dont-discover": [
              "intervention/image",
              "folklore/image"
            ]
        }
    },
```

**Comment by salehasadi** [comment]
> I found a slightly weird solution but it works(laravel 5.7) :)
> Change "extra' in your composer.json:
> 
> ```
> "extra": {
>        "laravel": {
>            "dont-discover": [
>              "intervention/image",
>              "folklore/image"
>            ]
>        }
>    },
> ```

Thanks man
